core/src/main/java/org/opensearch/sql/expression/aggregation/AvgAggregator.java

```diff
@@ -23,20 +32,39 @@
  */
 public class AvgAggregator extends Aggregator<AvgAggregator.AvgState> {
 
+  /**
+   * To process by different ways different data types, we need to store the type.
+   * Input data has the same type as the result.
+   */
+  private final ExprCoreType dataType;
+
   public AvgAggregator(List<Expression> arguments, ExprCoreType returnType) {
     super(BuiltinFunctionName.AVG.getName(), arguments, returnType);
+    dataType = returnType;
   }
 
   @Override
   public AvgState create() {
-    return new AvgState();
+    switch (dataType) {
+      case DATE:
```

Comment:
Well, meaning is almost the same as `avg(datetime)` or `avg(time)` has - something like to find average age or something/somebody.
I added this because:
* OpenSearch supports `avg(date)` aggregation, but we don't always pushdown aggregations to the engine. In that case we have to do this in memory.
* MySQL supports it and we're trying to align with it too.

There is no storable `INTERVAL` type in OpenSearch, and SQL has no function which returns it. There is no combination of query-dataset which will involve `avg(interval)`.

Reply:
As it was discussed outside of GH, it was decided to proceed with this feature.